kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -4339,6 +4339,20 @@ protected void removeResource(ContentResourceEdit edit, boolean removeContent) t
 		// htripath -store the metadata information into a delete table
 		// assumed uuid is not null as checkExplicitLock(id) throws exception when null
 
+
+		//check if resource is of type CitationsList and clean up citation tables
+		if (removeContent) {
+			if( edit.getResourceType().equals(CITATIONS_RESOURCE_TYPE_ID)) {
+				try {
+					String data = new String(edit.getContent());
```

Comment:
I see that your checking the resource type for it to be a citations list which is good, but adding the whole content of a file seems risky is there maybe a possibility of just add the specific data from the file? I ask only because if some large amount of data found it's way into a event it could potentially be harmful?

Reply:
Looks like this should only be a collection id. I am wondering if this is really exploitable. Could a user upload files and set arbitrary content types?

Reply:
Yes, inside the file there is only the collectionId. I can't see how a non admin user could exploit this. I dont think a user can set the Resource Type. The File Type of a citation list is already 'text/plain'. 

I can think of the follwing way to check if the content is a valid uuid. If the String data is too long an IllegalArgumentException is thrown.
```

String data = new String(edit.getContent());
try {
	java.util.UUID.fromString(data).toString();
	eventTrackingService.post(eventTrackingService.newEvent(CITATIONS_HARD_DELETE_EVENT, data, true));
} catch (IllegalArgumentException e) {
	log.warn("Invalid UUID from citation list {}", e);
}
```

Reply:
I wouldn't be worried about it then. @ern What do you think?

Reply:
if it should be a collection id then it would be simple to just verify that by calling `isCollection()`

Reply:
The isCollection() method only seems to check if String ends with "/". Should I update my PR?

Reply:
yeah I would agree with you about the isCollection being brain-less but I still think you should use it, maybe one day isCollection will do a better check.

Reply:
isCollection(ref) would return false, because the collectionId does not end with "/". 

it would need something like this: isCollection(data.endsWith("/") ? data : data + "/"); ?